lib/edgehog_device_forwarder/connections.ex

```diff
@@ -0,0 +1,70 @@
+# Copyright 2023 SECO Mind Srl
+# SPDX-License-Identifier: Apache-2.0
+
+defmodule EdgehogDeviceForwarder.Connections do
+  use Supervisor
+
+  alias EdgehogDeviceForwarder.Connections.TokenData
+
+  @tokens Application.compile_env!(:edgehog_device_forwarder, EdgehogDeviceForwarder.Connections)[
+            :tokens
+          ]
+
+  @type token :: String.t()
+  @type device_id :: TokenData.device_id()
+  @type realm :: TokenData.realm()
+  @type socket :: TokenData.socket()
+
+  # API
+
+  ## Token
+  @spec register_device_info(token, %{:device_id => device_id, :realm => realm}) ::
+          :ok | {:error, :token_not_found}
+  def register_device_info(token, device_info) do
+    ConCache.update(@tokens, token, &TokenData.put_device_info(&1, device_info))
+  end
+
+  @spec register_socket(token, socket) :: :ok | {:error, :already_exists}
+  def register_socket(token, socket) do
+    # TODO: check if we need to handle socket's process termination ourself.
+    # I'd be inclined to say no from
+    #   - https://hexdocs.pm/phoenix/Phoenix.Socket.Transport.html#c:terminate/2
+    #   - https://hexdocs.pm/phoenix/Phoenix.Socket.Transport.html#module-custom-transports
+    ConCache.insert_new(@tokens, token, TokenData.new(socket))
+  end
+
+  @spec close_socket(token) :: :ok
+  def close_socket(token) do
+    ConCache.delete(@tokens, token)
+  end
+
+  @spec fetch_socket(token) :: {:ok, socket} | {:error, :token_not_found}
+  def fetch_socket(token) do
+    case ConCache.get(@tokens, token) do
+      nil -> {:error, :token_not_found}
+      %TokenData{socket: socket} -> {:ok, socket}
+    end
+  end
+
+  # Callbacks
+
+  def start_link(init_arg) do
+    Supervisor.start_link(__MODULE__, init_arg, name: __MODULE__)
+  end
+
+  @impl Supervisor
+  def init(_init_args) do
+    [
+      con_cache_child(@tokens)
+    ]
+    |> Supervisor.init(strategy: :one_for_one)
+  end
+
+  defp con_cache_child(name) do
+    params = [name: name, ttl_check_interval: false]
+    id = {ConCache, name}
+
+    {ConCache, params}
+    |> Supervisor.child_spec(id: id)
+  end
```

Comment:
I don't get why you need all of this (it's an extra Supervisor that just supervises one child).
You can just put
```elixir
...
      EdgehogDeviceForwarderWeb.Endpoint,
      {ConCache, name: :tokens, ttl_check_interval: false}
...
```
in the Application supervision tree

Reply:
Because
1. we add a second `ConCache` child in #8 
2. it's separation of concerns, the cache module supervises the caches